lib/max.cpp

```diff
@@ -0,0 +1,137 @@
+#include "flag_gems/operators.h"
+#include "flag_gems/utils.h"
+
+#include <iostream>
+#include "c10/cuda/CUDAStream.h"
+#include "torch/torch.h"
+#include "triton_jit/triton_jit_function.h"
+
+#include <filesystem>
+#include "ATen/WrapDimUtils.h"
+#include "ATen/native/ReduceOpsUtils.h"
+#include "c10/util/DimVector.h"
+
+namespace {
+std::tuple<at::Tensor, int64_t, int64_t> permute_reduction_axes_right(
+    const at::Tensor &tensor, at::OptionalIntArrayRef reduction_axes_opt) {
```

Comment:
You can simplify 的 reduction_axes to an integer, since we are implementing these 2 overloads. (Although there are more overloads for `max`).

Check `aten/src/ATen/native/native_functions.yaml` for more details.

- func: max(Tensor self) -> Tensor
- func: max.dim(Tensor self, int dim, bool keepdim=False) -> (Tensor values, Tensor indices)

Reply:
Actually torch has a `structured delegate` mechanism to delegate `max.dim` to `max.dim_max` where the output values and indices are passed in as output parameters. We may also implement that.